frontend/src/modules/Environments/components/EnvironmentRedshiftConnectionAddForm.js

```diff
@@ -0,0 +1,365 @@
+import { GroupAddOutlined } from '@mui/icons-material';
+import { LoadingButton } from '@mui/lab';
+import {
+  Autocomplete,
+  Box,
+  CardContent,
+  CircularProgress,
+  Dialog,
+  Divider,
+  Grid,
+  MenuItem,
+  TextField,
+  Typography
+} from '@mui/material';
+import { Formik } from 'formik';
+import { useSnackbar } from 'notistack';
+import PropTypes from 'prop-types';
+import React from 'react';
+import * as Yup from 'yup';
+import { SET_ERROR, useDispatch } from 'globalErrors';
+import { useClient, useFetchGroups } from 'services';
+import { createRedshiftConnection } from '../services';
+
+export const EnvironmentRedshiftConnectionAddForm = (props) => {
+  const { environment, onClose, open, reload, ...other } = props;
+  const { enqueueSnackbar } = useSnackbar();
+  const dispatch = useDispatch();
+  const client = useClient();
+
+  let { groupOptions, loadingGroups } = useFetchGroups(environment);
+
+  const clusterOptions = [
```

Comment:
maybe use the new query introduced in #1435?

Reply:
Added in # todo list in https://github.com/data-dot-all/dataall/issues/955